### PR TITLE
Update to latest html5ever changes.

### PR DESCRIPTION
**Pull request by jdm**
Also includes deprecation fixes and an upgrade to the 2018 edition, along with updates to cssparser and selectors.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,20 +1,21 @@
 [package]
 name = "kuchiki"
-version = "0.7.3"
+version = "0.8.0"
```
Comment (SimonSapin) [inline Cargo.toml:3]: Does this PR include breaking changes? If I remember correctly all dependencies updated here are private.

**Comment by jdm** [comment]
I believe I checked and it does.

**Comment by Ygg01** [comment]
@jdm is there a tool that helps with figuring out if semver needs to be updated?

**Comment by jdm** [comment]
https://github.com/rust-dev-tools/rust-semverver

File: Cargo.toml (hunk 1 above)
Comment (Ygg01) [reply Cargo.toml:3]: I believe there are breaking changes to src/select.rs#207 related to changing version of CSS selectors

Before we had `local_name(&self) -> &LocalName`, now there is a `has_local_name(&self, &LocalName) -> bool`

**Comment by Ygg01** [comment]
Looks good to me. And I believe SimonSapin's concerns are addressed.

File: Cargo.toml (hunk 1 above)
Comment (SimonSapin) [reply Cargo.toml:3]: But that’s an impl of a trait from a private dependency. It’s not exposed to users (or at least it’s not intended to).

Comment (jdm) [reply Cargo.toml:3]: https://github.com/kuchiki-rs/kuchiki/blob/62110940e56b9ef462e294087ae07c057c45382f/src/tree.rs#L54

NodeData is exposed through the public API, and QualName comes from the public html5ever API.

Comment (SimonSapin) [reply Cargo.toml:3]: Ah, good point. Looks like I either remembered wrong about html5ever being a private dependency, or made that mistake from the beginning.

Comment (Ygg01) [reply Cargo.toml:3]: @SimonSapin should a new issue for this be raised (having html5ever as a public dependency)?

**Pull request merged**